bitsv/network/services/network.py

```diff
@@ -138,33 +147,17 @@ def get_balance(self, address):
         :raises ConnectionError: If all API services fail.
         :rtype: ``int``
         """
-
-        for api_call in self.GET_BALANCE:
-            try:
-                return self.retry_wrapper_call(api_call, address)
-            except IGNORED_ERRORS:
-                self.list_of_apis.rotate(-1)  # failed api --> back of the que for next time (across all api calls)
```

Comment:
From the existing code, It looks self.GET_BALANCE will not change after we call list_of_apis.rotate(-1). 

Therefore I think the logic is not working in the existing code?

Reply:
Yes. **Right now**, rotating the deque does nothing because we only have **one API in the deque**. But it's setup ready for adding in more... That's why I refactored it that way.

This reduces the chances of mixing and matching APIs across e.g. getting utxos and broadcasting... But we can do the retry stuff on each API before we decide that it has failed. Then --> rotate deque.

So order for API A, B and C is:
1) Retry logic API A
2) if fail --> Rotate order of APIs for remainder of the session
3) Retry logic API B
4) if fail --> Rotate order of APIs for remainder of the session
5) Retry logic API C
if fail --> Rotate order of APIs for remainder of the session + **All APIs have failed...**